field/VerticalStaggerLoc.F90

```diff
@@ -0,0 +1,95 @@
+module mapl3g_VerticalStaggerLoc
+   implicit none
+   private
+
+   public :: VerticalStaggerLoc
+   public :: VERTICAL_STAGGER_NONE
+   public :: VERTICAL_STAGGER_EDGE
+   public :: VERTICAL_STAGGER_CENTER
+   public :: VERTICAL_STAGGER_INVALID
+
+   public :: operator(==)
+   public :: operator(/=)
+
+   public :: make_VerticalStaggerLoc
+
+   type :: VerticalStaggerLoc
+      private
+      integer :: id
+   contains
+      ! TODO: Convert to DTIO once compilers support allocatable internal files
+      procedure :: to_string
+   end type VerticalStaggerLoc
```

Comment:
Could you move the name string into the derived type?
```suggestion
   type :: VerticalStaggerLoc
      private
      integer :: id
      character(len=:), allocatable :: name
   contains
      ! TODO: Convert to DTIO once compilers support allocatable internal files
      procedure :: to_string
   end type VerticalStaggerLoc
```

Reply:
This would allow more easy lookup by name string.

Reply:
Unfortunately, this would prevent declaring constants (parameters).   This is an annoying Fortran feature that you can't use a structure constructor to fill in an allocatable string, and at the same time you cannot use a user-define constructor in a constant expression.

I think the priority is for having defined constants. Hence the implementation here.